instrumentation/spring/spring-boot-actuator-autoconfigure-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/actuator/AutoConfigurationImportSelectorInstrumentation.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.actuator;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.returns;
+
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import java.util.ArrayList;
+import java.util.List;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+public class AutoConfigurationImportSelectorInstrumentation implements TypeInstrumentation {
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("org.springframework.boot.autoconfigure.AutoConfigurationImportSelector");
+  }
+
+  @Override
+  public void transform(TypeTransformer transformer) {
+    transformer.applyAdviceToMethod(
+        named("getCandidateConfigurations").and(returns(List.class)),
+        getClass().getName() + "$GetCandidateConfigurationsAdvice");
+  }
+
+  @SuppressWarnings("unused")
+  public static class GetCandidateConfigurationsAdvice {
+
+    @Advice.OnMethodExit(suppress = Throwable.class)
+    public static void onExit(@Advice.Return(readOnly = false) List<String> configurations) {
+      if (configurations.contains(
+          "org.springframework.boot.actuate.autoconfigure.metrics.MetricsAutoConfiguration")) {
+        List<String> configs = new ArrayList<>(configurations.size() + 1);
+        configs.addAll(configurations);
+        // using class reference here so that muzzle will consider it a dependency of this advice
+        configs.add(OpenTelemetryMeterRegistryAutoConfiguration.class.getName());
```

Comment:
I can't remember, why is it important for this to be a _dependency_ of the advice? it seems like we wouldn't need to inject it as a helper class since it is loaded via `ClassPathResource`

Reply:
Oh, it's for the reference collection, without it muzzle wouldn't capture the referenced spring & micrometer classes. I'll add a comment here.